rest/src/main/groovy/whelk/rest/api/ESQueryLensBoost.groovy

```diff
@@ -0,0 +1,121 @@
+package whelk.rest.api
+
+import whelk.JsonLd
+
+/**
+ * A utility to compute a list of boosted fields for ElasticSearch.
+ * The fields are computed from lens definitions of chips ans cards. It is
+ * currently contingent on the JsonLd.SEARCH_KEY mechanism to avoid the
+ * repetition of nested fields.
+ */
+class ESQueryLensBoost {
+
+    JsonLd jsonld
+
+    ESQueryLensBoost(JsonLd jsonld) {
+        this.jsonld = jsonld
+    }
+
+    List<String> computeBoostFieldsFromLenses(String[] types) {
+        def boostFields = []
+        def seenKeys = [] as Set
+
+        def chipsLenses = jsonld.displayData.lensGroups?.chips
+        def cardsLenses = jsonld.displayData.lensGroups?.cards
+
+        boostFields += [ "$JsonLd.SEARCH_KEY^100" as String ]
+
+        def baseTypes = ['Identity', 'Instance', 'Item']
+
+        // Compute boost for "chip" properties.
+        def selectedChipsLenses = selectLenses(
+                chipsLenses, types, baseTypes, seenKeys)
+
+        boostFields += selectedChipsLenses.sum { lens ->
+            int boost = 200
+            collectBoostFields(lens, boost, seenKeys)
+        }
+
+        // Compute boost for "card" properties.
+        def selectedCardsLenses = selectLenses(
+                cardsLenses, types, baseTypes, seenKeys)
+
+        boostFields += selectedCardsLenses.sum { lens ->
```

Comment:
This closure is a bit gnarly. Consider splitting it up/extracting it like with `collectBoostFields`.

Reply:
Certainly!